frontend/src/metabase/admin/performance/components/StrategyEditorForDatabases.tsx

```diff
@@ -0,0 +1,328 @@
+import type { Location } from "history";
+import { useCallback, useEffect, useRef, useState } from "react";
+import type { InjectedRouter, Route } from "react-router";
+import { withRouter } from "react-router";
+import { push } from "react-router-redux";
+import { useAsync } from "react-use";
+import { t } from "ttag";
+import { findWhere, pick } from "underscore";
+import type { SchemaObjectDescription } from "yup/lib/schema";
+
+import { useDatabaseListQuery } from "metabase/common/hooks";
+import LoadingAndErrorWrapper from "metabase/components/LoadingAndErrorWrapper";
+import { useConfirmation } from "metabase/hooks/use-confirmation";
+import { color } from "metabase/lib/colors";
+import { useDispatch } from "metabase/lib/redux";
+import { PLUGIN_CACHING } from "metabase/plugins";
+import { CacheConfigApi } from "metabase/services";
+import { Box, Stack } from "metabase/ui";
+
+import { rootId } from "../constants";
+import { useDelayedLoadingSpinner } from "../hooks/useDelayedLoadingSpinner";
+import { useRecentlyTrue } from "../hooks/useRecentlyTrue";
+import type {
+  Config,
+  SafelyUpdateTargetId,
+  Strategy,
+  StrategyType,
+} from "../types";
+import { Strategies } from "../types";
+
+import { ResetAllToDefaultButton } from "./ResetAllToDefaultButton";
+import { Panel, TabWrapper } from "./StrategyEditorForDatabases.styled";
+import { StrategyForm } from "./StrategyForm";
+import { StrategyFormLauncher } from "./StrategyFormLauncher";
+
+export const StrategyEditorForDatabases = withRouter(
+  ({
+    canOverrideRootStrategy,
+    router,
+    route,
+  }: {
+    canOverrideRootStrategy: boolean;
+    router: InjectedRouter;
+    route?: Route;
+  }) => {
```

Comment:
I think I'd like to see this component split up a bit personally. We have 204 lines of code before we get to the actual returned mark up, which is a sign to me that this is probably doing too much.

Reply:
Thanks for the nudge - I've now split it into more components and hooks